Changes to the compute capabilities to account for the two different Blackwell Edge GPUs.

dimvar opened the pull request:
RTX PRO 6000 has CC 12.0.
Spark has CC 12.1.

Removed the IsAtLeastBlackwellPro method because there is no guarantee that future data center GPUs will have CC higher than 12.0.

Also skipped the latency estimator test on Edge GPUs because it uses the collective performance model and crashes here:
https://github.com/openxla/xla/blob/784702574ee3f2df06116e7f7e4b837150c8694a/xla/service/gpu/model/gpu_collective_performance_model.cc#L239

Artem-B commented on xla/stream_executor/cuda/cuda_compute_capability.h:
```diff
@@ -205,7 +187,15 @@ struct CudaComputeCapability {
 
   bool IsBlackwellPro() const {
-    return major == CudaComputeCapabilities::kBlackwellPro;
+    return major == CudaComputeCapabilities::kBlackwellEdge && minor == 0;
+  }
+
+  bool IsBlackwellSpark() const {
+    return major == CudaComputeCapabilities::kBlackwellEdge && minor == 1;
+  }
+
+  bool IsEdgeGpu() const {
+    return major == CudaComputeCapabilities::kBlackwellEdge;
```
Naming -- one of the hardest unsolved problems in computer science. :-) 

> we shouldn't do isBlackwell(enum variant)

Fair enough. I don't have a strong preference on this, just speculating what those functions are intended to provide. We generally do not care about card variants, but do care about GPU variants/CC. If we already have helpers for checking compute capability, do we need product-name-based helpers at all? I'm mostly curious. If they are useful, I'm fine with that.

> The page you already linked to is the only one I know.

It tends to lag behind the GPU architecture variants exposed by CUDA SDK. That would be one more reason to stick with just CC/sm_XXX nomenclature, IMO.

dimvar replied:
> do we need product-name-based helpers at all?

For the test I'm fixing, I just need "IsEdgeGpu".

I was assuming Google needs IsBlackwellPro somewhere, that's why it got added. But I'm happy to delete it and also IsBlackwellRtx if they're not needed yet.

dimvar committed:
Some Spark fixes.                                                                                                                                                                                                                                                             Rename kBlackwellPro to kBlackwell12, as the sm_12x compute capabilities also include Spark.

Fix the latency estimator test and the gemm fusion autotuner test for Spark.

Removed the IsAtLeastBlackwellPro method because there is no guarantee that future
data center GPUs will have CC higher than 12.0.
